src/main/scala/tech/cryptonomic/nautilus/cloud/domain/authentication/AuthorizationService.scala

```diff
@@ -16,4 +17,26 @@ object AuthorizationService {
       f.map(Right(_))
     else
       AccessDenied(requiredRole, session).asLeft[T].pure[F]
+
+  def requiredRoleOrApiKey[F[_]: Applicative, T](
+      requiredRole: Role,
+      apiKey: Option[String],
+      supportedKeys: List[String]
```

Comment:
I'd suggest using sets for membership's checks, they should be better tailored to the task.

Reply:
Ok, changed. Just for the sake of argument - in theory it's a a good idea to perform membership checks, but the time saved by this change won't ever surpass the time I spent writing this comment 😛